guides/source/active_storage_overview.md

```diff
@@ -382,138 +138,97 @@ class SignupController < ApplicationController
-
 [`has_one_attached`]: https://api.rubyonrails.org/classes/ActiveStorage/Attached/Model.html#method-i-has_one_attached
 [Attached::One#attach]: https://api.rubyonrails.org/classes/ActiveStorage/Attached/One.html#method-i-attach
 [Attached::One#attached?]: https://api.rubyonrails.org/classes/ActiveStorage/Attached/One.html#method-i-attached-3F
 
 ### `has_many_attached`
 
-The [`has_many_attached`][] macro sets up a one-to-many relationship between records
-and files. Each record can have many files attached to it.
+The [`has_many_attached`][] method sets up a one-to-many relationship between a
+record and attached files. Each record can have many files attached to it.
 
-For example, suppose your application has a `Message` model. If you want each
-message to have many images, define the `Message` model as follows:
+For example, suppose your application has a `Product` model. Each product can have multiple images associated with it:
 
 ```ruby
-class Message < ApplicationRecord
+class Product < ApplicationRecord
   has_many_attached :images
 end
 ```
 
-or if you are using Rails 6.0+, you can run a model generator command like this:
+You can also use the model generator command like this:
 
 ```bash
-$ bin/rails generate model Message images:attachments
+$ bin/rails generate model Product images:attachments
 ```
 
-You can create a message with images:
+The controller to create a product with multiple images looks like this:
 
 ```ruby
-class MessagesController < ApplicationController
+class ProductsController < ApplicationController
   def create
-    message = Message.create!(message_params)
-    redirect_to message
+    product = Product.create!(product_params)
+    redirect_to product
   end
 
   private
-    def message_params
-      params.expect(message: [ :title, :content, images: [] ])
+    def product_params
+      params.expect(product: [ :title, :content, images: [] ])
     end
 end
 ```
 
-Call [`images.attach`][Attached::Many#attach] to add new images to an existing message:
+You can call [`images.attach`][Attached::Many#attach] to add new images to an existing product:
 
 ```ruby
-@message.images.attach(params[:images])
+@product.images.attach(params[:images])
 ```
 
-Call [`images.attached?`][Attached::Many#attached?] to determine whether a particular message has any images:
+You can call [`images.attached?`][Attached::Many#attached?] to determine whether a particular product has any images:
 
 ```ruby
-@message.images.attached?
+@product.images.attached?
 ```
 
-Overriding the default service is done the same way as `has_one_attached`, by using the `service` option:
+NOTE: When using `has_many_attached`, calling `images.attach(...)` adds new attachments. It does not replace or overwrite existing ones. If you want to replace existing images, you must explicitly [purge](#removing-files) the old attachments before attaching new ones.
```

Comment:
Further one we also use `append` which maybe is a bit less ambiguous
```suggestion
NOTE: When using `has_many_attached`, calling `images.attach(...)` appends new attachments. It does not replace or overwrite existing ones. If you want to replace existing images, you must explicitly [purge](#removing-files) the old attachments before attaching new ones.
```
Another option could be:
```suggestion
NOTE: When using `has_many_attached`, calling `images.attach(...)` adds new attachments to the existing attachments. It does not replace or overwrite existing ones. If you want to replace existing images, you must explicitly [purge](#removing-files) the old attachments before attaching new ones.
```